compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -929,6 +933,77 @@ fn compare_synthetic_generics<'tcx>(
     if error_found { Err(ErrorReported) } else { Ok(()) }
 }
 
+fn compare_const_generic_types<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    impl_m: &ty::AssocItem,
+    trait_m: &ty::AssocItem,
+    trait_item_span: Option<Span>,
+) -> Result<(), ErrorReported> {
+    let const_params_of = |def_id| {
+        tcx.generics_of(def_id).params.iter().filter_map(|param| match param.kind {
+            GenericParamDefKind::Const { .. } => Some(param.def_id),
+            _ => None,
+        })
+    };
+    let const_params_impl = const_params_of(impl_m.def_id);
+    let const_params_trait = const_params_of(trait_m.def_id);
+
+    for (const_param_impl, const_param_trait) in iter::zip(const_params_impl, const_params_trait) {
+        let impl_ty = tcx.type_of(const_param_impl);
+        let trait_ty = tcx.type_of(const_param_trait);
+        if impl_ty != trait_ty {
+            let (impl_span, impl_ident) = match tcx
+                .hir()
+                .find(tcx.hir().local_def_id_to_hir_id(const_param_impl.expect_local()))
```

Comment:
```suggestion
                .get_if_local(const_param_impl)
```
or just `get(tcx.hir().local_def_id_to_hir_id(const_param_impl.expect_local()))`. I didn't know `get_if_local` existed and don't like that method too much but :shrug:

Reply:
Fixed. Thanks for the review!